app/models/product_import/product_importer.rb

```diff
@@ -100,7 +100,8 @@ def entries_json
         entries[entry.line_number] = {
           attributes: entry.displayable_attributes,
           validates_as: entry.validates_as,
-          errors: entry.invalid_attributes
+          errors: entry.invalid_attributes,
+          product_validations: entry.product_validations
```

Comment:
I know it's not strictly related to this PR, but IMO the JSON serialization should rather be extracted into a serializer. We could create an issue for it, if we all agree.

Reply:
Issue added: https://github.com/openfoodfoundation/openfoodnetwork/issues/3105